src/Controllers/SignUpController.php

```diff
@@ -30,12 +28,9 @@ public function __invoke(Request $request, Response $response, ?array $args = []
         $email = $request->getParam('email');
         $crypt  = $request->getParam('password');
         $password = password_hash($crypt, PASSWORD_DEFAULT);
-
-        $user = new User($username, $email, $password);
-
-        $this->em->persist($user);
-        $this->em->flush();
-
-        return $this->twig->render($response, 'registro.twig');
+        
+        $useCase = new useCase($username,$email,$password);
```

Comment:
Recuerda que hemos inyectado el caso de uso al controlador, por lo tanto esto está mal. Lo que se tiene que hacer es llamar al caso de uso que hemos asignado a la propiedad `useCase` y tal y como se está haciendo aquí es erróneo, ya que estás construyendo el caso de uso con los parámetros que se le pasan al invoke.

Lo correcto sería hacerlo así `$this->useCase->__invoke($username,$email,$password)`.

Por otra parte fíjate que estás haciendo una instancia con `new` de algo llamado `useCase` que es una clase que no existe, por lo tanto daría error